Coding/fw/lib/WebServer/CaptivePortal.cpp

```diff
@@ -0,0 +1,172 @@
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+***************************************************************************************************/
+/**
+@addtogroup Webserver
+@{
+@file       CaptivePortal.cpp
+
+Captive Portal for ESP32 WebServer @ref CaptivePortal.h
+
+* @}
+***************************************************************************************************/
+/* INCLUDES ***************************************************************************************/
+#include <CaptivePortal.h>
+#include <Web_Config.h>
+#include <SPIFFS.h>
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* CONSTANTS **************************************************************************************/
+
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+static void reqRestart();                                    /*< Changes the value of restartRequested to True*/
+static void credentialsProcessor(String name, String value); /*< Processor to save the required credentials */
+
+/**
+* Captive portal request handler.
+*/
+class CaptiveRequestHandler : public AsyncWebHandler
+{
+public:
+    /**
+    * Constructs the captive portal request handler.
+    */
+    CaptiveRequestHandler()
+    {
+    }
+
+    /**
+    * Destroys the captive portal request handler.
+    */
+    ~CaptiveRequestHandler()
+    {
+    }
+
+    /**
+    * Checks whether the request can be handled.
+    * @param[in] request   Web request
+    * @return If request can be handled, it will return true otherwise false.
+    */
+    bool canHandle(AsyncWebServerRequest *request) override
+    {
+        /* The captive portal handles every request. */
+        return true;
+    }
+
+    /**
+    * Handles the request.
+    * @param[in] request   Web request, which to handle.
+    */
+    void handleRequest(AsyncWebServerRequest *request) override
+    {
+        if (nullptr == request)
+        {
+            return;
+        }
+        /** Forces authentication */
+        if (!request->authenticate(WebConfig::getWEB_USER().c_str(), WebConfig::getWEB_PASS().c_str()))
+        {
+            return request->requestAuthentication();
+        }
+
+        if (HTTP_POST == request->method())
+        {
+            if (request->args() != 0)
+            {
+                int params = request->params();
+                for (int i = 0; i < params; i++)
+                {
+                    AsyncWebParameter *p = request->getParam(i);
+                    credentialsProcessor(p->name(), p->value());
+                }
+                request->send(SPIFFS, "/setCredentials.html", String(), false, captivePageProcessor);
```

Comment:
Try to store strings that are part of the interface to the html world in one place.  It is hard to find all instances of such strings later on, if they are scattered over the sources. 
Proposal:

 const char * htmpPages[] = 
{
       "/STACredentials.html",
       "/foo/bar.html"
}

enum HmtlPageIds { HTML_PAGE_CREDENTIALS, HTML_PAGE_FOO ...}

then use htmlPages[HTML_PAGE_CREDENTIALS] in the invokation below.